discovery_engine_core/web-api/openapi.yaml

```diff
@@ -0,0 +1,153 @@
+openapi: 3.0.3
+info:
+  title: Personalization API
+  version: 1.0.0-rc1
+  description: |-
+    The endpoints can be contacted at `t<name>.xaynet.dev/v1` or `t<name>.xaynet.dev/v1.0`.
+    The `v1` path will always point to the latest `v1.x`.
+    
+    # Auth
+    To authenticate with the API two bearer tokens will be provided. One will work for the
+    personalization endpoints and the other one will allow to add documents to the system.
+    
+tags:
+  - name: personalization
+    description: Interaction of the user and personalized document
+  - name: ingestion
+    description: Add documents to the system
+paths:
+  /users/{user_id}/documents/personalized:
+    get:
+      tags:
+        - personalization
+      summary: Get personalized document for the user
+      description: |-
+        Returns a map from `document_id` to the score of that document for the
+        specified `user_id`. The score is a value between 0 and 1 where an higher value
+        means that the document matches more the preferences of the user.
+      parameters:
+        - name: user_id
+          in: path
+          description: Id of the user
+          required: true
+          schema:
+            $ref: '#/components/schemas/UserId'
+        - name: count
+          in: query
+          description: Maximum number of personalized documents to return
+          type: integer
+          format: int32
+          minimum: 1
+          maximum: 100
+      responses:
+        '200':
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/PersonalizedDocuments'
+        '400':
+          description: invalid user id
+        '404':
+          description: user not found or the user does not have enough interactions
+
+  /users/{user_id}/documents/interacted:
```

Comment:
Because you can only report a single document this should be probably named 
`users/{user_id}/document/interacted`

Also much simpler would be: 

GET `users/{user_id}/document/{documentId}/interacted`

Reply:
It will be easier to extend in the future if the user wants to report more then one documents with one call time.
Even you report only one document the collections is still `documents`, especially in the case where you put the `{documentId}` in the path.